crypto/pem/pem_pkey.c

```diff
@@ -75,6 +75,10 @@ static EVP_PKEY *pem_read_bio_key_decoder(BIO *bp, EVP_PKEY **x,
         }
     ERR_pop_to_mark();
 
+    /* if we were asked for private key, the public key is optional */
+    if (selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY)
+        selection = selection & ~OSSL_KEYMGMT_SELECT_PUBLIC_KEY;
+
```

Comment:
Does it ensure it's optional?

Reply:
We clear the bit when calling the has() check so yeah, this ensures we do not check the OSSL_KEYMGMT_SELECT_PUBLIC_KEY bit if we are already checking for the OSSL_KEYMGMT_SELECT_PRIVATE_KEY bit.